tests/test-update-install-flatpaks.c

```diff
@@ -896,27 +896,30 @@ test_update_install_through_squashed_list (EosUpdaterFixture *fixture,
 
   etc_data_init (data, fixture);
 
+  /* XXX: We can't seem to use G_N_ELEMENTS for getting sub-array sizes
+   * reliably below, they all seem to just return 3 */
```

Comment:
Drop this comment, since that’s what `G_N_ELEMENTS` is expected to do: `G_N_ELEMENTS(A)` is defined as `sizeof (A) / sizeof (*A)`, and the declaration of `flatpaks_to_install` explicitly says the `sizeof (*A) == 3`. Hard-coding the numbers in the calls as you’ve done below seems fine to me.

Reply:
Ah, I was referring to:

    G_N_ELEMENTS(flatpaks_to_install[0])

Which should return `1` but instead returns `3` (`flatpaks_to_install[0]` has a `sizeof` 96, according to gdb)

Reply:
But in any event I'll drop the comment. Just seemed counterintuitive to me.

Reply:
Ah right, now I see what you mean. I remember now that I had to hardcode a sub-array size of `3` since the compiler couldn't work out what size to give the parent array otherwise. And I suppose that will mean it always get an allocation of 96 bytes.

Makes sense (and no longer counterintuitve). I'll drop the comment.